File: pset1-C/mario/mario2.c
#include <stdio.h>
#include <cs50.h>

// Função get_altura
int get_height_int(void)
{
    // Declarar variável no escopo global
    int n;
    do {
        n = get_int("Height: ");
    } while (n < 1 || n > 8);
    return n;
}

//Função para imprimir pontos na horizontal
void print_dots(int dots) {
    for (int z = 0; z < dots; z++) {
        printf(".");
    }
}

// Função principal
int main(void)
{
    int height = get_height_int();
    for (int i = 1; i <= height; i++) {
        // Imprime os pontos do lado esquerdo
        print_dots(height - i);

        for (int j = 0; j < i; j++) {
            // Imprime os hashes do lado esquerdo
            printf("#");
        }

        // Espaço entre as duas colunas de hashes
        printf("  ");

        for (int j = 0; j < i; j++) {
            // Imprime os hashes do lado direito
            printf("#");
        }

        // Imprime os pontos do lado direito
        print_dots(height - i);

        //Gera uma nova linha após cada iteração
        printf("\n");
    }
    return 0;
}

File: pset6 - Python/mario/mario2.py
from cs50 import get_int

while True:
    n = get_int("Height: ")
    if n > 0 and n < 9:
        break

for i in range(0, n, 1):
    for j in range(0, n+i+3, 1):
        if (j==n or j==n+1 or i+j < n-1):
            print(" ", end="")
        else:
            print("#", end="")
    print()
